protocol/da-footprint-block-limit.md

```diff
@@ -0,0 +1,299 @@
+more detailed analysis of the same problem and context and a similar solution.
+
+## Proposed Solution
+
+A _DA footprint block limit_ is introduced to limit the total amount of estimated compressed transaction data that can
+fit into a block. The base fee update rules take this new total DA footprint into account, so the base fee market isn't
+broken like for policy-level throttling solutions.
+
+The main idea is to introduce a new _DA footprint_ resource that is tracked next to gas, and limit it by the same
+block gas limit. This is inspired by the
+[Multidimensional Gas Metering](https://ethresear.ch/t/a-practical-proposal-for-multidimensional-gas-metering/22668)
+design, but the regular gas isn't touched, like in this design for L1 Ethereum.
+
+Similarly to the regular gas usage of transaction calldata, we calculate a transaction’s _DA footprint_ by taking
+its RLP-encoding, calculate the compressed transaction DA usage estimation from
+[Fjord](https://specs.optimism.io/protocol/fjord/exec-engine.html#fjord-l1-cost-fee-changes-fastlz-estimator)
+and multiply it by a factor, the _DA footprint gas scalar_, to get to the transaction’s DA footprint. 
+
+```python
+da_usage_estimate = max(minTransactionSize, intercept + fastlzCoef*fastlzSize / 1e6)
+da_footprint = da_usage_estimate * da_footprint_gas_scalar
+```
+
+Here, `fastlzSize` is the length of the FastLZ-compressed RLP-encoding of a transaction.
+
+Now when evaluating whether a transactions still fits into a block, we take the maximum over the two resources, regular
+total gas use of all transactions, and new total DA footprint, and see if this max is still below the block gas limit.
+The Multidimensional Gas Metering design introduces a new block field `gas_metered` to differentiate it from `gas_used`
+as the sum total of all transaction's gas used. However, it is proposed to just repurpose a block's `gas_used` field to
+hold the maximum over both resources' totals:
+
+
+```python
+block.gas_used = max(sum(tx.gas_used), sum(tx.da_footprint if tx.type != DepositTxType))
+```
+
+A block's total DA footprint only takes non-Deposit transactions into account, because Deposit transactions aren't part
+of batches, so they don't contribute to a block's DA usage.
+
+This definition has the advantage that the base fee update mechanics just keep working, now taking into account high
+total block DA footprint.
+
+The DA footprint cost is comparable to the calldata floor cost. If it's set to a higher value, say 400, you
+can expect _roughly_ a 1/10 of incompressible calldata to fit into a block compared to only limiting it by
+transaction gas usage alone. If the block is mostly low-calldata transactions that use much more regular gas than DA
+footprint, this new mechanism wouldn’t have any effect. This is the case for the vast majority of blocks.
+
+### DA footprint gas scalar
+
+I propose to make the DA footprint gas scalar parameter configurable via the `SystemConfig`. The next Ethereum hardfork
```

Comment:
Perhaps clarify _who_ can change the configuration, i.e. `SystemConfigOwner`.

Reply:
done